crates/client/devnet/src/lib.rs

```diff
@@ -325,6 +335,162 @@ mod tests {
         DevnetForTesting { backend, contracts, block_production, mempool }
     }
 
+    #[rstest]
+    #[case("../../../cairo/target/dev/madara_contracts_TestContract.contract_class.json")]
+    fn test_erc_20_declare(mut chain: DevnetForTesting, #[case] contract_path: &str) {
+        println!("{}", chain.contracts);
+
+        let sender_address = &chain.contracts.0[0];
+
+        let sierra_class: SierraClass = serde_json::from_reader(std::fs::File::open(contract_path).unwrap()).unwrap();
+        let flattened_class: FlattenedSierraClass = sierra_class.clone().flatten().unwrap();
+
+        // starkli class-hash target/dev/madara_contracts_TestContract.compiled_contract_class.json
+        let compiled_contract_class_hash =
+            Felt::from_hex("0x0138105ded3d2e4ea1939a0bc106fb80fd8774c9eb89c1890d4aeac88e6a1b27").unwrap();
+
+        let declare_txn: BroadcastedDeclareTransaction =
+            BroadcastedDeclareTransaction::V3(BroadcastedDeclareTransactionV3 {
+                sender_address: sender_address.address,
+                compiled_class_hash: compiled_contract_class_hash,
+                signature: vec![],
+                nonce: Felt::ZERO,
+                contract_class: Arc::new(flattened_class),
+                resource_bounds: ResourceBoundsMapping {
+                    l1_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                    l2_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                },
+                tip: 0,
+                paymaster_data: vec![],
+                account_deployment_data: vec![],
+                nonce_data_availability_mode: DataAvailabilityMode::L1,
+                fee_data_availability_mode: DataAvailabilityMode::L1,
+                is_query: false,
+            });
+
+        let res = chain.sign_and_add_declare_tx(declare_txn, sender_address);
+
+        let calculated_class_hash = sierra_class.clone().class_hash().unwrap();
+
+        assert_eq!(res.class_hash, calculated_class_hash);
+
+        chain.block_production.set_current_pending_tick(1);
+        chain.block_production.on_pending_time_tick().unwrap();
+
+        let block = chain.backend.get_block(&BlockId::Tag(BlockTag::Pending)).unwrap().unwrap();
+
+        assert_eq!(block.inner.transactions.len(), 1);
+        assert_eq!(block.inner.receipts.len(), 1);
+        log::info!("receipt: {:?}", block.inner.receipts[0]);
+
+        let class_info =
+            chain.backend.get_class_info(&BlockId::Tag(BlockTag::Pending), &calculated_class_hash).unwrap().unwrap();
+        match class_info {
+            ClassInfo::Sierra(info) => {
+                assert_eq!(info.compiled_class_hash, compiled_contract_class_hash);
+            }
+            ClassInfo::Legacy(_) => {}
+        }
+
+        let TransactionReceipt::Declare(receipt) = block.inner.receipts[0].clone() else { unreachable!() };
+
+        assert_eq!(
```

Comment:
same as above

Reply:
Here we don't need assert matches as we are comparing the receipt.